_posts/2019-10-26-dividing-human-pie-ai.md

```diff
@@ -0,0 +1,59 @@
+
+AWS provides some great example exercises that demonstrate the capabilities of [Amazon SageMaker](https://aws.amazon.com/sagemaker/), [Amazon Lex](https://aws.amazon.com/lex/), [Amazon Personalize](https://aws.amazon.com/personalize/) and [Amazon Forecast](https://aws.amazon.com/forecast/). In the afternoon session participants worked through a choice of these exercises at their own pace while a colleague and I offered help and explanation if they got stuck.
+
+Feedback from a recent event indicated that participants would've liked more guidance in the hands-on session. We're always keen to try new things, figure out what works and what doesn't, fix things quickly and continuously improve. However, opinions were split on how to improve the experience for participants.
+
+* Some participants would have preferred more explanation of how the AWS cloud works, and that they'd have appreciated working more slowly when configuring the infrastructure.
+
+* Some developers felt that we'd discussed the data science aspects too briefly, so they didn't have time to pick up any new knowledge in that area.
+
+* Another respondent said that the machine learning content could have been more challenging, because we'd focused on the basics.
+
+## How to divide a human pie?
+Dividing the participants into two groups would give the participants more contact time with the helpers. The helpers would also have some data to help them to tailor their guidance to address their group's knowledge gaps. Asking for help always feels easier in smaller groups, too. 
+
+But given the participants' diversity of experience, how should we divide the groups to give everyone a better experience?
+
+There were so many possible ways to divide our human pie that there was no clear answer. I needed some data!
+
+During a morning break I asked all of the participants to rate themselves in terms of their machine learning knowledge, where on the developer - data scientist spectrum they identify themselves, and how much cloud experience they had, using a Google Form.
+
+I fed the responses into a Python app that used the [_scikit-learn_](https://scikit-learn.org/) library's [_k_-means clustering algorithm](https://scikit-learn.org/stable/modules/clustering.html#k-means) to assign the responses into two groups. 
+
+_k_-means is a relatively simple unsupervised learning algorithm that can be used to group similar data points together. 
+
+It starts by randomly choosing locations for _k_ cluster centroids, where _k_ is the desired number of groups. The distances to the data points are then calculated, and data points are each assigned to their nearest centroid. The algorithm then moves each centroid to reduce the sum of the distances between the data points and their respective centroids.
```

Comment:
Could you plot what this stage of the cluster looks like (dumb folk like me like pretty pictures)

Reply:
I can't get access to that information from scikit-learn, however I could visualise the final positions of the centroids?